src/pal/src/numa/numa.cpp

```diff
@@ -527,6 +554,103 @@ SetThreadGroupAffinity(
 
 /*++
 Function:
+  SetThreadAffinityMask
+
+See MSDN doc.
+--*/
+DWORD_PTR
+PALAPI
+SetThreadAffinityMask(
+  IN HANDLE hThread,
+  IN DWORD_PTR dwThreadAffinityMask
+)
+{
+    PERF_ENTRY(SetThreadAffinityMask);
+    ENTRY("SetThreadAffinityMask(hThread=%p, dwThreadAffinityMask=%p)\n", hThread, dwThreadAffinityMask);
+
+    CPalThread *pCurrentThread = InternalGetCurrentThread();
+    CPalThread *pTargetThread = NULL;
+    IPalObject *pTargetThreadObject = NULL;
+
+    PAL_ERROR palErr =
+        InternalGetThreadDataFromHandle(pCurrentThread, hThread,
+                                        0, // THREAD_SET_CONTEXT
+                                        &pTargetThread, &pTargetThreadObject);
+
+    if (NO_ERROR != palErr)
+    {
+        ERROR("Unable to obtain thread data for handle %p (error %x)!\n", hThread,
+              palErr);
+        return 0;
+    }
+
+    pthread_t thread = pTargetThread->GetPThreadSelf();
+
+#if HAVE_PTHREAD_GETAFFINITY_NP
+    cpu_set_t prevCpuSet;
+    CPU_ZERO(&prevCpuSet);
+    KAFFINITY prevMask = 0;
+
+    int st = pthread_getaffinity_np(thread, sizeof(cpu_set_t), &prevCpuSet);
+
+    if (st == 0)
+    {
+        for (int i = 0; i < g_possibleCpuCount; i++)
+        {
+            if (CPU_ISSET(i, &prevCpuSet))
+            {
+                prevMask |= ((KAFFINITY)1) << i;
+            }
+        }
+    }
+
+    cpu_set_t cpuSet;
+    CPU_ZERO(&cpuSet);
+
+    int cpu = 0;
+    while (dwThreadAffinityMask)
+    {
+        if (dwThreadAffinityMask & 1)
+        {
+            CPU_SET(cpu, &cpuSet);
+        }
+        cpu++;
+        dwThreadAffinityMask >>= 1;
+    }
+
+    st = pthread_setaffinity_np(thread, sizeof(cpu_set_t), &cpuSet);
+
+    if (st == -1)
```

Comment:
I haven't checked this myself yet, but are you sure this is the right way to handle errors for this function? Based on the man pages I'm seeing (for example, [this one](https://linux.die.net/man/3/pthread_setaffinity_np)), it looks like this function just returns the error code directly, so you'd just check for a nonzero return value and switch based on that value rather than `errno`.